### DISCUSSION TITLE
Calls between getters with typescript

### DISCUSSION THREAD

**Discussion by alex129** [vuejs/pinia#1987] [Help and Questions]
Hello,

In my store I have splitted getters in different files, this is not a problem but I have to make calls between my getters but typescript show me an error and to disable the error I have to put 'as unknown as T'

 These are my getters:

![image](https://user-images.githubusercontent.com/10452211/217467083-b1317525-60e8-420a-8d9f-54f7a319986d.png)

And the error is:

any
Property 'forEach' does not exist on type '(state: CoreState) => Material[]'.

The problem es that when I call to other getter inside a getter, the typescript detects that the other getter is a function.

Any idea?

**Comment by Lilja** [comment]
Typescript and classes written with vue doesn't work great. At least that's is what I experienced when moving from vue 2 to vue 3. I would perhaps look at creating a store with the composition api.
